clone/internal/verify/verify.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package verify
+
+import (
+	"fmt"
+
+	"github.com/google/trillian/merkle/compact"
+)
+
+type StreamLeaves func(start, end uint64, out chan []byte, errc chan error)
```

Comment:
Could probably do with a comment here - the "in order-ness" bit in the comment on the DB method which satisfies this signature seems like it's part of the contract here?

Reply:
Pretty shocking that I missed all of these. You did the right thing calling it out, this was an oversight.